packages/web3-eth/src/eth_tx.ts

```diff
@@ -0,0 +1,315 @@
+			// transaction's number properties have been converted to HexStrings.
+			// JavaScript doesn't handle negative hex strings e.g. -0x1, but our
+			// numberToHex method does. -0x1 < 0 would result in false, so we must check if
+			// hex string is negative via the inclusion of -
+			transaction.gas === undefined ||
+			transaction.gasPrice === undefined ||
+			transaction.gas.startsWith('-') ||
+			transaction.gasPrice.startsWith('-')
+		)
+			throw new InvalidGasOrGasPrice({
+				gas: transaction.gas,
+				gasPrice: transaction.gasPrice,
+			});
+	} else if (
+		transaction.maxFeePerGas === undefined ||
+		transaction.maxPriorityFeePerGas === undefined ||
+		transaction.maxFeePerGas.startsWith('-') ||
+		transaction.maxPriorityFeePerGas.startsWith('-')
+	)
+		throw new InvalidMaxPriorityFeePerGasOrMaxFeePerGas({
+			maxPriorityFeePerGas: transaction.maxPriorityFeePerGas,
+			maxFeePerGas: transaction.maxFeePerGas,
+		});
+
+	const hasEip1559 =
+		transaction.maxFeePerGas !== undefined || transaction.maxPriorityFeePerGas !== undefined;
+	if (transaction.gasPrice !== undefined && (transaction.type === '0x2' || hasEip1559))
+		throw new Eip1559GasPriceError(transaction.gasPrice);
+	if ((transaction.type === '0x0' || transaction.type === '0x1') && hasEip1559)
+		throw new UnsupportedFeeMarketError({
+			maxFeePerGas: transaction.maxFeePerGas,
+			maxPriorityFeePerGas: transaction.maxPriorityFeePerGas,
+		});
+};
+
+export const validateTransactionForSigning = (
+	transaction: Transaction,
+	overrideMethod?: (transaction: Transaction) => void,
+) => {
+	if (overrideMethod !== undefined) {
+		overrideMethod(transaction);
+		return;
+	}
+
+	if (typeof transaction !== 'object' || transaction === null)
+		throw new InvalidTransactionObjectError(transaction);
+
+	const formattedTransaction = formatTransaction(transaction, ValidTypes.HexString);
+
+	validateCustomChainInfo(transaction);
+	validateChainInfo(transaction);
+	validateGas(formattedTransaction);
+
+	if (
+		formattedTransaction.nonce === undefined ||
+		formattedTransaction.chainId === undefined ||
+		formattedTransaction.nonce.startsWith('-') ||
+		formattedTransaction.chainId.startsWith('-')
+	)
+		throw new InvalidNonceOrChainIdError({
+			nonce: transaction.nonce,
+			chainId: transaction.chainId,
+		});
+};
+
+export async function populateTransaction<
+	DesiredType extends ValidTypes,
+	ReturnType = ValidReturnTypes[DesiredType],
+>(
+	transaction: Transaction,
+	web3Context: Web3Context<EthExecutionAPI>,
+	desiredType: DesiredType,
+	privateKey?: HexString,
+	overrideMethod?: (
+		transaction: Transaction,
+		web3Context: Web3Context<EthExecutionAPI>,
+	) => PopulatedUnsignedTransaction<ReturnType>,
+): Promise<PopulatedUnsignedTransaction<ReturnType>> {
+	if (overrideMethod !== undefined) return overrideMethod(transaction, web3Context);
+
+	const populatedTransaction = { ...transaction };
+	const web3Eth = new Web3Eth(web3Context.currentProvider);
```

Comment:
I think for scalability requirement, its more good not to create new `Web3Eth` object on every call inside this function.

Reply:
Yea I agree, this was a quick and dirty way to get it done. Some alternatives I can think of:

1. Import the needed methods from [rpc_methods.ts](https://github.com/ChainSafe/web3.js/blob/4.x/packages/web3-eth/src/rpc_methods.ts) - this way the user only needs to pass [web3Context](https://github.com/ChainSafe/web3.js/blob/4.x/packages/web3-core/src/web3_context.ts), and we pass the `web3Context.requestManager` to the `rpc_method`. However, we do loose the benefit of input validation and automatic output formatting since those are only defined in `Web3Eth` (the reason I went initing `Web3Eth`)
2. Export each `Web3Eth` method (basically just already exported `rpc_methods.ts` with input validation and output formatting, plus some alias methods like [getBlockTransactionCount](https://github.com/ChainSafe/web3.js/blob/5a00ab5b4098654280811bf098030e7db485441c/packages/web3-eth/src/index.ts#L151) that use provided input to figure out what `rpc_method` to call), and then have `Web3Eth` include them as class methods - I think this makes more sense than 1., users/devs still have access to use the bare bones implementation (`rpc_methods.ts`) or the validated and formatted implementation

Reply:
Is there any limitation of not passing object of Web3Eth to this function?

Reply:
For now, I don't think so. The `populateTransaction` method just needs access to `Web3Context`, so might as well only require that instead of an entire `Web3Eth` class instance that may not be available when this method is imported